src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisor.scala

```diff
@@ -0,0 +1,69 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import akka.actor.SupervisorStrategy.Restart
+import akka.actor.{Actor, ActorRef, OneForOneStrategy, Props}
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.leonardo.config.MonitorConfig
+import org.broadinstitute.dsde.workbench.leonardo.dao.DataprocDAO
+import org.broadinstitute.dsde.workbench.leonardo.db.DbReference
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, ClusterRequest}
+import org.broadinstitute.dsde.workbench.leonardo.monitor.ClusterMonitorSupervisor._
+import org.broadinstitute.dsde.workbench.leonardo.service.LeonardoService
+
+object ClusterMonitorSupervisor {
+  def props(monitorConfig: MonitorConfig, gdDAO: DataprocDAO, dbRef: DbReference): Props =
+    Props(new ClusterMonitorSupervisor(monitorConfig, gdDAO, dbRef))
+
+  sealed trait ClusterSupervisorMessage
+  case class RegisterLeoService(service: LeonardoService) extends ClusterSupervisorMessage
+  case class ClusterCreated(cluster: Cluster) extends ClusterSupervisorMessage
+  case class ClusterDeleted(cluster: Cluster, recreate: Boolean = false) extends ClusterSupervisorMessage
+  case class RecreateCluster(cluster: Cluster) extends ClusterSupervisorMessage
+}
+
+class ClusterMonitorSupervisor(monitorConfig: MonitorConfig, gdDAO: DataprocDAO, dbRef: DbReference) extends Actor with LazyLogging {
+
+  var leoService: LeonardoService = _
+
+  override def receive: Receive = {
+    case RegisterLeoService(service) =>
+      leoService = service
+
+    case ClusterCreated(cluster) =>
+      logger.info(s"Monitoring cluster ${cluster.googleProject}/${cluster.clusterName} for initialization.")
+      startClusterMonitorActor(cluster)
+
+    case ClusterDeleted(cluster, recreate) =>
+      logger.info(s"Monitoring cluster ${cluster.googleProject}/${cluster.clusterName} for deletion.")
+      startClusterMonitorActor(cluster, recreate)
+
+    case RecreateCluster(cluster) =>
+      if (monitorConfig.recreateCluster) {
+        logger.info(s"Recreating cluster ${cluster.googleProject}/${cluster.clusterName}...")
+        val clusterRequest = ClusterRequest(cluster.googleBucket, cluster.googleServiceAccount, cluster.labels)
+        leoService.createCluster(cluster.googleProject, cluster.clusterName, clusterRequest)
```

Comment:
What happens if this call fails?

Reply:
Ehm, good catch. This could probably use some error handling. I'm not sure we should keep retrying if the _recreate_ cluster fails, but we should at least be logging an error.

Reply:
(Currently I think it would fail silently because the future is not awaited on)